src/autotype/AutoTypeMatchModel.cpp

```diff
@@ -45,6 +45,20 @@ QModelIndex AutoTypeMatchModel::indexFromMatch(const AutoTypeMatch& match) const
     return index(row, 1);
 }
 
+QModelIndex AutoTypeMatchModel::closestIndexFromMatch(const AutoTypeMatch& match) const
+{
+    int row = -1;
+
+    for (int i = 0; i < m_matches.size(); i++) {
+        const auto& currentMatch = m_matches.at(i);
+        if (currentMatch.first == match.first || (row == -1 && currentMatch.first == match.first)) {
+            row = i;
+        }
```

Comment:
You should add a ``break`` here to avoid unnecessary iterations and perhaps use STL iterators.